[Index Management] Implement index details page

yuliacech opened the pull request:
## Summary
This PR removes the feature flag and enables the new index details page by default. The index details page was implemented in following PRs: 
- https://github.com/elastic/kibana/pull/163521
- https://github.com/elastic/kibana/pull/163955
- https://github.com/elastic/kibana/pull/164741
- https://github.com/elastic/kibana/pull/165027
- https://github.com/elastic/kibana/pull/165038
- https://github.com/elastic/kibana/pull/165456

In this PR we completely remove now obsolete code for the old index details flyout: react components and corresponding redux code. All related tests are updated and cleaned up. The config value for Index Management plugin `xpack.index_management.dev.enableIndexDetailsPage` is deprecated as unused and can be removed in v9.0.


### How to test
1. Start ES and Kibana with `yarn es snapshot` and `yarn start`
3. Navigate to Index Management and create an index
4. Click the index name in the table and check the tabs of the details page
### Screenshots
#### Stateful
Overview 
<img width="1387" alt="Screenshot 2023-09-27 at 14 41 57" src="https://github.com/elastic/kibana/assets/6585477/e58b15e7-d10c-4473-873c-d0f128392404">


Mappings
<img width="1392" alt="Screenshot 2023-09-27 at 14 42 05" src="https://github.com/elastic/kibana/assets/6585477/441157cb-5a26-47c3-8da0-b4df51ebec5d">


Settings 
<img width="1385" alt="Screenshot 2023-09-27 at 14 42 13" src="https://github.com/elastic/kibana/assets/6585477/da66a2eb-1f21-44c1-9356-484c66caab88">


Statistics
<img width="1380" alt="Screenshot 2023-09-27 at 14 42 22" src="https://github.com/elastic/kibana/assets/6585477/ec93d85c-e754-4c21-88ab-0124dc114fc9">


Error loading data
<img width="1333" alt="Screenshot 2023-09-26 at 19 05 37" src="https://github.com/elastic/kibana/assets/6585477/fc1804b3-6aa0-4019-bae6-e7bb40113b28">
<img width="1327" alt="Screenshot 2023-09-26 at 19 06 07" src="https://github.com/elastic/kibana/assets/6585477/ca711697-cc74-4ba8-b17c-ec9b01f3026e">
<img width="1329" alt="Screenshot 2023-09-26 at 19 06 28" src="https://github.com/elastic/kibana/assets/6585477/0cb46b09-8542-452a-8845-40d060057e95">
<img width="1331" alt="Screenshot 2023-09-26 at 19 06 48" src="https://github.com/elastic/kibana/assets/6585477/87de8d3d-b6e5-4e8f-b27c-18a1c6e950d8">


Error saving index settings
<img width="1332" alt="Screenshot 2023-09-26 at 19 07 31" src="https://github.com/elastic/kibana/assets/6585477/e6e4b3d0-c237-4d0a-995a-4562bc78f88e">


### Serverless
Overview 
<img width="1336" alt="Screenshot 2023-09-26 at 19 51 47" src="https://github.com/elastic/kibana/assets/6585477/6c76c23b-4be6-4ab3-ae1d-c7ae751e100d">


Mappings
<img width="1336" alt="Screenshot 2023-09-26 at 19 23 51" src="https://github.com/elastic/kibana/assets/6585477/625fa703-506f-4389-9df0-86441a655074">


Settings
<img width="1332" alt="Screenshot 2023-09-26 at 19 24 02" src="https://github.com/elastic/kibana/assets/6585477/c496ab09-f2db-4c1b-9fb6-1e9b64b1c142">


# Release note
Index details can now be viewed on a new index details page in Index Management. 
<img width="1387" alt="Screenshot 2023-09-27 at 14 41 57" src="https://github.com/elastic/kibana/assets/6585477/b90c706d-8b15-49e4-8f6a-cb66f3ed1822">

mdefazio commented:
Adding this comment here so as to not clutter the [other PR](https://github.com/elastic/kibana/pull/164628).

(Referencing the bulk action menu)
> I'm not sure it makes sense to keep those "show" actions with the new page for index details

It might be worthwhile to keep these.  Currently these are one-click into the flyout tab depending on the choice—perhaps worth considering maintaining that expectation of the user if they are looking for a specific action. (With the added value that these are now a dedicated view). As opposed to them not finding an action they were used to selecting.

yuliacech commented:
Hey @mdefazio, thanks a lot for your feedback! I put the index actions back except for "Edit settings" since this is now in the same tab as "show settings".
<img width="544" alt="Screenshot 2023-09-22 at 18 34 27" src="https://github.com/elastic/kibana/assets/6585477/b0fd4907-4e02-4abf-8607-191329ba466d">

elasticmachine commented:
Pinging @elastic/platform-deployment-management (Team:Deployment Management)

mdefazio commented:
@yuliacech Thank you!

kc13greiner reviewed (approved):
LGTM!!

mdefazio reviewed (approved):
LGTM

gchaps commented:
Here are some suggestions for improving the copy in the Index management UI. I'm happy to iterate with you on the text.

## Overview page

- In Discover, we use `Field statistics`.  We should be consistent and use `Statistics` for the tab name. 
- Text 
  - Uses both "add data" and "add documents".  Pick one word and be consistent.  
  - Shorten the title. 
  - In the description
    - End the sentence with a period. 
    - Include the name of the API and link to the doc if it adds value

Here is a suggestion:

> Add data to this index
> Use the _name_ API to add documents to your index.  Learn more.

<img width="545" alt="Screenshot 2023-09-25 at 12 37 15 PM" src="https://github.com/elastic/kibana/assets/33642766/322f86b2-93d5-4fbd-8fd5-62452ac21bb7">

## Loading index data

- Use this format for the error.  Be sure to capitalize URL. 

> Unable to load index details
>
>We encountered an error loading data for index _name_. Make sure that the index name in the URL is correct and try again.
>
>Reload

<img width="438" alt="Screenshot 2023-09-25 at 12 43 21 PM" src="https://github.com/elastic/kibana/assets/33642766/61be8697-f2c9-42c0-8eed-aea6ba534c14">

## Index actions

- Toasts are currently in the form of "Successfully closed _index-name_.   [The EUI guidelines for toasts](https://eui.elastic.co/pr_3350/#/guidelines/toasts) recommends this format.

> The index _index-name_ was closed.  

## Statistics tab

- In the `About index stats` description:
  - Change `while` to `and`
  
<img width="547" alt="Screenshot 2023-09-25 at 1 09 31 PM" src="https://github.com/elastic/kibana/assets/33642766/524c54aa-3af3-49e2-97bd-ace034387740">

- I think the error message was already updated, but just in case:

> Unable to load index stats
>
>We encountered an error loading the stats for index _name_.
>
>Reload

- Can you please provide a screenshot of the empty state? I'm wondering if something like this might work:

>**No index stats to display yet**
>After you open your index, you'll be able to view the stats here.

gchaps commented:
@mdefazio Each tab in this UI has a Learn more link.  Is there a way we can include the information inside a popover (or somewhere else in the UI) so the user doesn't have to leave the product to get help?

yuliacech commented:
Thanks a lot for your review, @gchaps! I've implemented your suggestions and updated the PR description with new screenshots.
To answer some of your questions: 
- I made all error callouts look the same. The most probable reason for a failing request is that the index name is incorrect. Other than that and trying reloading the request, I'm not sure that we can offer any other guidance. 
- When an index is closed, we still want to display the "Statistics" tab to keep the UI consistent. Once the index is opened again, the "Statistics" tab can be viewed again. The index status can be easily changed via "Manage index" button. 
- When the settings can't be saved, the message is coming directly from ES. I've adjusted the formatting to make it clearer.

yuliacech commented:
Thanks a lot for the review, @alisonelizabeth! Yes, I still need to update the "Release note" section in the PR description. I will copy the screenshots with the reviewed copy there.

gchaps commented:
Small nit:

Maybe for the mappings and stats tabs, the doc link should also be "Learn more about xxxx."
